sim/midas/src/main/scala/midas/core/CPUManagedStreamEngine.scala

```diff
@@ -201,33 +201,34 @@ class CPUManagedStreamEngine(p: Parameters, val params: StreamEngineParameters)
     genCRFile()
 
     override def genHeader(base: BigInt, sb: StringBuilder) {
-      val headerWidgetName = getWName.toUpperCase
       super.genHeader(base, sb)
 
-      def serializeStreamParameters(prefix: String, params: Seq[StreamDriverParameters]): Unit = {
+      def serializeStreamParameters(name: String, params: Seq[StreamDriverParameters]): Unit = {
         val numStreams = params.size
-        sb.append(genConstStatic(s"${headerWidgetName}_${prefix}_stream_count", UInt32(numStreams)))
-
-        // Hack: avoid emitting a zero-sized array by providing a dummy set of
-        // parameters when no streams are generated. This is a limitation of the
-        // current C emission strategy. Note, the actual number of streams is still reported above.
-        val placeholder = StreamDriverParameters("UNUSED", 0, 0, 0)
-        val nonEmptyParams = if (numStreams == 0) Seq(placeholder) else params
-
-        val arraysToEmit = Seq(
-          "names"         -> nonEmptyParams.map { p => CStrLit(p.name) },
-          "dma_addrs"     -> nonEmptyParams.map { p => UInt64(p.bufferBaseAddress) },
-          "count_addrs"   -> nonEmptyParams.map { p => UInt64(base + p.countMMIOAddress) },
-          "buffer_sizes"  -> nonEmptyParams.map { p => UInt32(p.bufferCapacity) },
-        )
-
-        for ((name, values) <- arraysToEmit) {
-          sb.append(genArray(s"${headerWidgetName}_${prefix}_${name}", values))
+        sb append s"std::vector<CPUManagedStreams::StreamParameters>{\n"
+        for (p <- params) {
+          sb append s"CPUManagedStreams::StreamParameters("
```

Comment:
Can new code avoid the use of infix notation:
`sb append <...> ` -> `sb.append(<...>)`